src/wp-admin/includes/media.php

```diff
@@ -3795,3 +3827,73 @@ function wp_media_attach_action( $parent_id, $action = 'attach' ) {
 		exit;
 	}
 }
+
+/**
+ * Outputs a select box to select upload media category.
+ *
+ * Displays only if media storage option has been set to 'category'.
+ *
+ * @since CP-2.1.0
+ */
+function cp_select_upload_media_category() {
+	$media_select = '';
+	$storefolders = get_option( 'uploads_use_yearmonth_folders' );
+	if ( $storefolders === '3' ) {
+		$cat_subfolder = get_option( 'media_cat_upload_folder' );
+
+		$media_terms = get_terms(
+			array(
+				'taxonomy'   => 'media_category',
+				'hide_empty' => false,
+			)
+		);
+
+		if ( ! empty( $media_terms ) ) {
+			$media_select .= '<label for="upload-category"><strong>' . __( 'Please choose the upload media category' ) . '</strong></label><br>';
+			$media_select .= '<select id="upload-category" name="upload-category">';
+			$media_select .= '<option value="">&nbsp;' . __( 'Media category' ) . '&nbsp;</option>';
```

Comment:
The `__()` function here should probably be wrapped in `esc_html()`.

Reply:
Escaping is necessary when it's a variable, so the value of the string is not known. But when it's known, as here, it serves no purpose.

Reply:
But in essence it is a variable as it can be translated, a malicious translation file could result in HTML injection and perhaps code execution.

Reply:
Fair enough! Latest commit fixes this.